packages/xstate-wallet/src/workflows/tests/conclude-channel.test.ts

```diff
@@ -0,0 +1,248 @@
+  const amountB1 = (await bStore.chain.getChainInfo(targetChannelId)).amount;
+
+  // store entries should have been udpated to finalized state
+  const entryA1 = await aStore.getEntry(targetChannelId);
+  const entryB1 = await bStore.getEntry(targetChannelId);
+  expect(entryA1.isFinalized).toBe(true);
+  expect(entryB1.isFinalized).toBe(true);
+
+  // Conclude again
+  await runUntilSuccess(concludeChannel);
+  const amountA2 = (await aStore.chain.getChainInfo(targetChannelId)).amount;
+  const amountB2 = (await bStore.chain.getChainInfo(targetChannelId)).amount;
+
+  const entryA2 = await aStore.getEntry(targetChannelId);
+  const entryB2 = await bStore.getEntry(targetChannelId);
+
+  expect(amountA2).toMatchObject(amountA1);
+  expect(amountB2).toMatchObject(amountB1);
+
+  // No change to the store entires, meaning that turnNum, etc. remain the same
+  expect(entryA1).toMatchObject(entryA2);
+  expect(entryB1).toMatchObject(entryB2);
+});
+
+it('can conclude again when A crashes during the first conclude attempt', async () => {
+  // Let A and B create and fund channel
+  await runUntilSuccess(createChannel);
+
+  // Simulate A crashes before withdrawing
+  const aMachine = interpret(concludeChannel(aStore).withContext(context))
+    .onTransition(state => {
+      if (state.value === 'withdrawing') {
+        aMachine.stop();
+      }
+    })
+    .start();
+
+  const entryA1 = await aStore.getEntry(targetChannelId);
+  expect(entryA1.isFinalized).toBe(false);
+
+  // A starts again
+  await new Promise(resolve =>
+    interpret(concludeChannel(aStore).withContext(context))
+      .start()
+      .onTransition(state => state.matches('success') && resolve())
+  );
+
+  const entryA2 = await aStore.getEntry(targetChannelId);
+  expect(entryA2.isFinalized).toBe(true);
+});
+
+it('reaches the same state when running conclude twice for virtual funding', async () => {
+  await createLedgerChannels();
+
+  // Both conclude the channel
+  await runUntilSuccess(concludeChannel, 'virtual');
+  const amountA1 = (await aStore.chain.getChainInfo(targetChannelId)).amount;
+  const amountB1 = (await bStore.chain.getChainInfo(targetChannelId)).amount;
+
+  // store entries should have been udpated to finalized state
+  const entryA1 = await aStore.getEntry(targetChannelId);
+  const entryB1 = await bStore.getEntry(targetChannelId);
+  expect(entryA1.isFinalized).toBe(true);
+  expect(entryB1.isFinalized).toBe(true);
+
+  // Conclude again
+  await runUntilSuccess(concludeChannel, 'virtual');
+  const amountA2 = (await aStore.chain.getChainInfo(targetChannelId)).amount;
+  const amountB2 = (await bStore.chain.getChainInfo(targetChannelId)).amount;
+
+  const entryA2 = await aStore.getEntry(targetChannelId);
+  const entryB2 = await bStore.getEntry(targetChannelId);
+
+  expect(amountA2).toMatchObject(amountA1);
+  expect(amountB2).toMatchObject(amountB1);
+
+  expect(outcomesEqual(entryA1.latestState.outcome, entryA2.latestState.outcome)).toBe(true);
```

Comment:
We should be able to expect the entry doesn't change.

Reply:
The issue is that the virtual-defunding-as-leaf workflow _also_ [concludes the target](https://github.com/statechannels/monorepo/blob/master/packages/xstate-wallet/src/workflows/virtual-defunding-as-leaf.ts#L38-L52) (in a less robust fashion)

Reply:
done.